mitosheet/mitosheet/tests/step_performers/test_merge.py

```diff
@@ -566,4 +566,38 @@ def test_delete_source_of_merged_sheet_no_optimizes():
     )
     mito.delete_dataframe(1)
 
-    assert len(mito.transpiled_code) > 0
\ No newline at end of file
+    assert len(mito.transpiled_code) > 0
+
+def test_merge_with_destination_sheet_indexes_the_same():
+    df1 = pd.DataFrame({'A': [1], 'B': [2]})
+    df2 = pd.DataFrame({'A': [1], 'B': [3]})
+    mito = create_mito_wrapper(df1, df2)
+    mito.merge_sheets(
+        'lookup', 0, 1, [['A', 'A']], ['A', 'B'], ['A', 'B'],
+    )
+    mito.merge_sheets(
+        'lookup', 0, 1, [['A', 'B']], ['A', 'B'], ['A', 'B'], destination_sheet_index=2
+    )
+    mito.merge_sheets(
+        'lookup', 0, 1, [['A', 'A']], ['A', 'B'], ['A', 'B'], destination_sheet_index=2
+    )
+
+    # There should be only be three dataframes
+    assert len(mito.dfs) == 3
```

Comment:
Can you check it is updated to the correct dataframe?

Reply:
Can you also check an update merge and then an undo?

Reply:
And finally, a merge, then an edit, then an update merge overwrites those steps.